### DISCUSSION TITLE
Executable flag setting

### DISCUSSION THREAD

**Discussion by GeneralAccess** [community/community#23172] [New to GitHub]
<p>I have a program file on GitHub which got there by clicking Add File&gt;Upload Files. It had the executable property on my PC, but the flag that I would expect to appear on the gh screen top left is not there. How can I set that flag?<br>
<img src="https://user-images.githubusercontent.com/20310498/181075429-41c57736-b122-4a68-be78-4d1a147c898f.png" alt="gh" data-base62-sha1="cWB7fXFmlOOPZ4FLwQqIAAZHpy2" width="451" height="329"></p>

**Comment by airtower-luna** [comment]
<p>On a Unixoid system (Linux, BSD, Mac, …) the executable flag being set differently should be shown as a difference, you can just <code>add</code> the file.</p>
<p>On Windows you can use <a href="https://git-scm.com/docs/git-add#Documentation/git-add.txt---chmod-x"><code>git add --chmod=+x FILE</code></a> to mark <code>FILE</code> as executable.</p>

**Comment by GeneralAccess** [comment]
<blockquote>
<p>the executable flag being set differently should be shown as a difference</p>
</blockquote>
<p>Sorry, I don’t understand that.</p>
<blockquote>
<p>you can just add the file</p>
</blockquote>
<p>I think you are talking about adding a file to a clone on my PC, is that correct? It’s a very simple application with only a couple of files, and I haven’t got a clone on my PC. I can’t be bothered with that extra complexity, or with learning how to do it. I can do everything I want to do without it, including building a Flatpak from it and submitting it to Flathub…apart from the one problem - the exec flag on the file…</p>
<p>(On Linux)</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Yes, I’m talking using the Git command line with a local repository, in both cases. I’m not aware of any way to set the flag over the web interface.</p>
<p>Theoretically you could try setting the flag in your build, commit, and push that back to your repository, but I’m pretty sure that’d be way more effort than just using the command line once.</p>

**Comment by GeneralAccess** [comment]
<p>Unfortunately I think you are right. I’ve found</p>
<p><a href="https://training.github.com/downloads/github-git-cheat-sheet/" rel="noopener nofollow ugc">https://training.github.com/downloads/github-git-cheat-sheet/</a><br>
and<br>
<a href="https://www.freecodecamp.org/news/learn-the-basics-of-git-in-under-10-minutes-da548267cc91/" rel="noopener nofollow ugc">https://www.freecodecamp.org/news/learn-the-basics-of-git-in-under-10-minutes-da548267cc91/</a></p>
<p>Which look helpful, I’ll have to bite the bullet and learn it. Thanks for your help anyway.<br>
[update]<br>
I’ve managed to set Git up on my PC, pull the github repo into it, change the executable file and push it back again. It now shows as executable. So that <em>was</em> the problem.</p>
<p>It all went wrong after that when I changed a file on gh and Git threw a wobbler - never mind I’ll get there in the end.<br>
Thanks again for your help.</p>

**Comment by GeneralAccess** [comment]
<p>I had been looking as some very poor explanations of how to use Git, once I found a clear explanation I realised that it wasn’t as complicated as I had thought. The Flatpak has now been published - thanks airtower-luna!</p>
